### ISSUE TITLE
exported generic types do not resolve their type parameters

### ISSUE THREAD

**Issue by sidharthv** [TypeStrong/typedoc#156]
For the below example:

``` javascript
declare interface Person {
    name: string;
}
declare type Z<T extends Person> = T & {
    address: string;
}   
declare interface IExample {
    someProperty: Z<Person>
}
```

The documentation shows:

``` javascript
someProperty: Z
```

and

``` javascript
Z: T & { address: string; }
```

How do I get it to show:

``` javascript
someProperty: Z<Person>
```

and something like:

``` javascript
Z: T & { address: string; }
T: Person
```

**Comment by sebastian-lenz** [comment]
Hi, thank you for your message. Type aliases are quite difficult to document as the compiler resolves them very early in the process so TypeDoc has to follow some thin hints to resolve them. I'll add this as an enhancement and will try to check back on it if I find the time.

**Comment by sidharthv** [comment]
Great, thanks!

Date: Tue, 8 Dec 2015 11:41:50 -0800

**Comment by Schniz** [comment]
Hey!
First of all, mad props about the product. This project is veryyyyyy good and produces high quality stuff! I'm impressed!

I wonder if the ~3 years that have passed have changed something in it in terms of TypeScript tooling? The following code has conditional types in it, but the TS server types it correctly; however, TypeDoc wouldn't "resolve" the type.

![image](https://user-images.githubusercontent.com/2054772/54264414-c0d52980-457b-11e9-88bc-661f462bc894.png)

This type is hand-rolled, but you can easily see how a `Pick` type (that comes with TypeScript) will act the same way on TS server and TypeDoc:

![image](https://user-images.githubusercontent.com/2054772/54264754-7607e180-457c-11e9-9716-7fb50e20ff75.png)

It is something that can be done? Is it something you are willing to support?
I believe that would make the docs much more verbose, but easier to read

**Comment by samtheprogram** [comment]
Came here looking for support for  `Pick` -- I have an interface like such (re-typed, no promise of 0 typos 😄):

```typescript
export interface RTPMediaExtensions extends Pick<RTCPeerConnection,
  | 'getSenders'
  | 'getReceivers'
  | 'getTransceivers'
  | 'addTrack'
  | 'removeTrack'
  | 'addTransceiver'
  | 'addEventListener'
  | 'removeEventListener'
  | 'ontrack'
> {}
```

With "Inherited" on, I am looking for the Interface's documentation to show these picked properties -- but instead it's an empty interface in typedoc's output. Not sure where to start looking in solving this though.

**Comment by Gerrit0** [comment]
Closing in favor of #1386 - this is fixed in 0.20.
